### PR TITLE
Improve GASNet co-locale performance tests

### PR DESCRIPTION
**Pull request by jhh67**
Improve GASNet co-locale performance tests by adding a test suite with PSHM disabled, and by clarifying that some `stream` and `indexgather` results are per-locale.

### CODE REVIEW

**Comment by bonachea** [comment]
Thanks for adding this @jhh67 ! 

I look forward to comparing the results once it reaches production.

File: util/cron/test-perf.hpe-apollo-hdr.gasnet-ibv.fast.colo-nopshm.bash [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+#
+# Run co-locale performance tests on an HPE Apollo over GASNet with PSHM
+# enabled.
```
[the remark this answers is not in the document]
Comment (jhh67) [reply util/cron/test-perf.hpe-apollo-hdr.gasnet-ibv.fast.colo-nopshm.bash:4]: Fixed.

File: util/cron/test-perf.hpe-apollo-hdr.gasnet-ibv.fast.colo-nopshm.bash [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+export CHPL_TEST_PERF_SUBDIR="hpe-apollo"
+export CHPL_TEST_PERF_CONFIG_NAME='16-node-apollo-hdr'
+
+source $CWD/common-perf.bash
+
+export CHPL_NIGHTLY_TEST_CONFIG_NAME="perf.hpe-apollo-hdr.gasnet-ibv.fast.colo-nopshm"
+
+source $CWD/common-hpe-apollo.bash
+source $CWD/common-perf-hpe-apollo-hdr.bash
+perf_hpe_apollo_args="-performance-configs gn-ibv-fast:v,gn-ibv-fast-colo:v,gasnet-ibv.fast.colo-nopshm:v -perflabel ml- -startdate 09/04/24"
```
[the remark this answers is not in the document]
Comment (jhh67) [reply util/cron/test-perf.hpe-apollo-hdr.gasnet-ibv.fast.colo-nopshm.bash:17]: Fixed.

**Review by jabraham17 — APPROVED** [review]

**Comment by bradcray** [comment]
Great, thanks John!

**Pull request merged**